gallery/src/DarkModeModal/DarkModeModal.tsx

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React, { useEffect, useState } from 'react';
+
+import {
+  NxButton,
+  NxModal,
+  NxForm,
+  NxFormGroup,
+  NxToggle,
+  useToggle,
+  NxFieldset,
+  NxRadio,
+  NxP,
+  NxCode
+} from '@sonatype/react-shared-components';
+import classnames from 'classnames';
+
+const DarkModeModal = () => {
+  const [showModal, setShowModal] = useState(false);
+  const modalCloseHandler = () => setShowModal(false);
+
+  const [enableModeChange, toggleEnableModeChange] = useToggle(false);
+  const [mode, setMode] = useState<string | null>(null);
+  const [radioDisabled, setRadioDisabled] = useState(true);
+
+  const htmlElement = document.documentElement;
+  const htmlClasses = classnames('nx-html', 'nx-html--page-scrolling',
+      {'nx-html--enable-color-schemes': enableModeChange,
+        'nx-html--light-mode': mode === 'light' && enableModeChange,
+        'nx-html--dark-mode': mode === 'dark' && enableModeChange
+      });
+
+  useEffect(()=> {
+    htmlElement.className = htmlClasses;
+  }, [enableModeChange, mode]);
+
+  useEffect(()=> {
+    enableModeChange ? setRadioDisabled(false) : setRadioDisabled(true);
```

Comment:
Whether the radiogroup is disabled is strictly based on `enableModeChange`, so you don't want `radioDisabled` to be its own piece of state at all – it can just be a local variable

Reply:
> Whether the radiogroup is disabled is strictly based on `enableModeChange`, so you don't want `radioDisabled` to be its own piece of state at all – it can just be a local variable

I updated it here : [a91181a](https://github.com/sonatype/sonatype-react-shared-components/pull/782/commits/a91181a9d5387764cb85f29a9d5bd4cb4b563dde). Since disabled is always the opposite of enableModeChange, I assigned the disabled attribute the opposite value of enableModeChange